src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -152,27 +159,178 @@ public void AddPackageReference(string projectPath, LibraryDependency libraryDep
             }
         }
 
+        /// <summary>
+        /// Add package version/package reference to the solution/project based on if the project has been onboarded to CPM or not.
+        /// </summary>
+        /// <param name="project">Project that needs to be modified.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        /// <param name="existingPackageReferences">Package references that already exist in the project.</param>
+        /// <param name="framework">Target Framework for which the package reference should be added.</param>
         private void AddPackageReference(Project project,
             LibraryDependency libraryDependency,
             IEnumerable<ProjectItem> existingPackageReferences,
             string framework = null)
         {
+            // Getting all the item groups in a given project
             var itemGroups = GetItemGroups(project);
 
-            if (!existingPackageReferences.Any())
+            // Add packageReference to the project file only if it does not exist.
+            var itemGroup = GetItemGroup(itemGroups, PACKAGE_REFERENCE_TYPE_TAG) ?? CreateItemGroup(project, framework);
+
+            if (!libraryDependency.VersionCentrallyManaged)
             {
-                // Add packageReference only if it does not exist.
-                var itemGroup = GetItemGroup(itemGroups, PACKAGE_REFERENCE_TYPE_TAG) ?? CreateItemGroup(project, framework);
-                AddPackageReferenceIntoItemGroup(itemGroup, libraryDependency);
+                if (!existingPackageReferences.Any())
+                {
+                    //Modify the project file.
+                    AddPackageReferenceIntoItemGroup(itemGroup, libraryDependency);
+                }
+                else
+                {
+                    // If the package already has a reference then try to update the reference.
+                    UpdatePackageReferenceItems(existingPackageReferences, libraryDependency);
+                }
             }
             else
             {
-                // If the package already has a reference then try to update the reference.
-                UpdatePackageReferenceItems(existingPackageReferences, libraryDependency);
+                // Get package version if it already exists in the props file. Returns null if there is no matching package version.
+                ProjectItem packageVersion = project.Items.LastOrDefault(i => i.ItemType == PACKAGE_VERSION_TYPE_TAG && i.EvaluatedInclude.Equals(libraryDependency.Name));
+                var versionCLIArgument = libraryDependency.LibraryRange.VersionRange.OriginalString;
```

Comment:
This function has not been changed from the original implementation since it is referring to updating package reference items when a project is not onboarded to CPM. So I believe it does not need to be aware if the version is passed from the command line or not.

Reply:
After discussing with @jeffkl and @kartheekp-ms I added a parameter to the AddPackageReference and related functions. This parameter is called PackageVersion and uses PackageReferenceArgs.PackageVersion. PackageReferenceArgs.PackageVersion is set based on if a version is passed as a CLI argument. This should resolve the issue.